app/src/main/java/com/github/se/icebreakrr/model/message/MeetingRequestViewModel.kt

```diff
@@ -399,54 +367,85 @@ class MeetingRequestViewModel(
    * the contact is too far away
    */
   fun meetingDistanceCancellation() {
-    val selfProfile = profilesViewModel.selfProfile.value
-    val originPoint = selfProfile?.location ?: GeoPoint(0.0, 0.0)
     updateInboxOfMessages {
       val contactUsers = profilesViewModel.getCancellationMessageProfile()
-      val distances =
-          contactUsers.map {
-            distanceBetweenGeoPoints(it.location ?: GeoPoint(0.0, 0.0), originPoint)
-          }
-      val mapUserDistance = contactUsers.zip(distances).toMap()
-      mapUserDistance.forEach {
-        if (it.value >= FIVE_HUNDRED_METERS_IN_KM) {
-          removeFromMeetingRequestInbox(it.key.uid)
-          removeFromMeetingRequestSent(it.key.uid) {}
-          val targetToken = it.key.fcmToken ?: "null"
-          val targetName = it.key.name
-          setMeetingCancellation(
-              targetToken, CancellationType.DISTANCE, targetName) // targeted to other user
-          sendMeetingCancellation()
-          setMeetingCancellation(senderToken, CancellationType.DISTANCE, it.key.name)
-          sendMeetingCancellation()
-        }
+      val usersInMessagingRange = profilesViewModel.getUsersInMessagingRange()
+      val contactUsersUid = contactUsers.map { it.uid }
+      val usersInMessagingRangeUid = usersInMessagingRange.map { it.uid }
+      val contactUserNotInRangeUid =
+          contactUsersUid.filter { !usersInMessagingRangeUid.contains(it) }
+      val contactUserNotInRange = contactUsers.filter { contactUserNotInRangeUid.contains(it.uid) }
+      contactUserNotInRange.forEach {
+        removeFromMeetingRequestInbox(it.uid)
+        removeFromMeetingRequestSent(it.uid) {}
```

Comment:
Same racing condition issues as in my other comment. The suggestion is the same. 

You can't get rid of all the racing conditions because you are making backend calls in a loop.

_Suggestion_ : make a function that can delete multiple uids in the same call to avoid racing conditions or just make one call to `updateProfile` with the updated inbox items

Reply:
I ll see what i can do here